addons/mail/models/mail_thread.py

```diff
@@ -2462,19 +2454,26 @@ def _notify_compute_recipients(self, message, msg_vals):
             exept_partner = [r['id'] for r in recipient_data['partners']]
             if author_id:
                 exept_partner.append(author_id)
-            sql_query = """ select distinct on (p.id) p.id from res_partner p
-                            left join mail_channel_partner mcp on p.id = mcp.partner_id
-                            left join mail_channel c on c.id = mcp.channel_id
-                            left join res_users u on p.id = u.partner_id
-                                where (u.notification_type != 'inbox' or u.id is null)
-                                and (p.email != ANY(%s) or p.email is null)
-                                and c.id = ANY(%s)
-                                and p.id != ANY(%s)"""
-
-            self.env.cr.execute(sql_query, (([email_from], ), (email_cids, ), (exept_partner, )))
-            for partner_id in self._cr.fetchall():
-                # ocn_client: will add partners to recipient recipient_data. more ocn notifications. We neeed to filter them maybe
-                recipient_data['partners'].append({'id': partner_id[0], 'share': True, 'active': True, 'notif': 'email', 'type': 'channel_email', 'groups': []})
+
+            new_pids = self.env['res.partner'].sudo().search([
+                ('id', 'not in', exept_partner),
+                ('channel_ids', 'in', email_cids),
+                ('email', 'not in', [email_from]),
+                '|',
+                ('user_ids', '=', False),
+                ('user_ids.notification_type', '!=', 'inbox'),
```

Comment:
Won"t that be translated into ("user_ids", "in", [A HUGE LIST OF IDS]) and be inefficient on a production db ?

Reply:
All users in the channel, but this is needed to be able to notify them... This function already return a big amount of partner in 13.0 so I guess it work on production database

Reply:
My concern is about the "('user_ids.notification_type', '!=', 'inbox')," part is likely to crash, because ORM will transform it into a user_ids in [...] with our 2M portal users, unless there is some auto_join magic involved.

Reply:
@tde-banana-odoo Hum indded, the SQL query executed is:

```
SELECT "res_partner".id FROM "res_partner" WHERE ((((("res_partner"."active" = true) AND (("res_partner"."id" not in (400)) OR "res_partner"."id" IS NULL)) AND ("res_partner"."id" in (SELECT "partner_id" FROM "mail_channel_partner" WHERE "channel_id" IN (254)))) AND (("res_partner"."email" not in ('e.e@example.com')) OR "res_partner"."email" IS NULL)) AND (("res_partner"."id" not in (SELECT "partner_id" FROM "res_users" where "partner_id" is not null)) OR ("res_partner"."id" in (SELECT "res_users"."partner_id" FROM "res_users" WHERE (("res_users"."active" = true) AND (("res_users"."notification_type" != 'inbox') OR "res_users"."notification_type" IS NULL)))))) ORDER BY  "res_partner"."display_name"
```

Which contains
```
SELECT "res_users"."partner_id" FROM "res_users" WHERE (("res_users"."active" = true) AND (("res_users"."notification_type" != 'inbox') OR "res_users"."notification_type" IS NULL))
```

Maybe should we keep the previous SQL query ?